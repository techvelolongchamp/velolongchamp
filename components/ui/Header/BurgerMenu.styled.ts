import styled from 'styled-components'

export const Container = styled.div`
  // for iOS Safari
  scroll-behavior: smooth;
  -webkit-overflow-scrolling: touch;
`

export const BurgerBars = styled.div<{ isBurgerMenuOpen: boolean }>`
  align-self: center;
  cursor: pointer;
  height: 22px;
  margin: 0 10px;
  min-width: 30px;
  transform: rotate(0deg);
  transition: 0.5s ease-in-out;
  width: 30px;

  & span {
    background: ${({ theme }) => theme.colors.grey};
    border-radius: 10px;
    height: 2px;
    left: 0;
    opacity: 1;
    position: absolute;
    transform: rotate(0deg);
    transition: 0.25s ease-in-out;
    width: 100%;

    &:nth-child(1) {
      top: 0;
    }
    &:nth-child(2),
    &:nth-child(3) {
      top: 10px;
    }
    &:nth-child(4) {
      top: 20px;
    }

    ${({ isBurgerMenuOpen }) =>
      isBurgerMenuOpen &&
      `
        &:nth-child(1), &:nth-child(4) {
            top: 10px;
            width: 0;
            left: 50%;
        }
        &:nth-child(2) { transform: rotate(45deg) }
        &:nth-child(3) { transform: rotate(-45deg) }
      `}
  }
`

export const BurgerContent = styled.div<{ isBurgerMenuOpen: boolean }>`
  position: absolute;
  transition: all 0.2s;
  top: 61px;
  left: ${({ isBurgerMenuOpen }) => (isBurgerMenuOpen ? '0' : '-100vw')};
  background: ${({ theme }) => theme.colors.grey};
  color: ${({ theme }) => theme.colors.white};
  width: 100vw;
  height: calc(100vh - 61px);
  overflow: auto;
`

export const LinkContainer = styled.ul`
  list-style: none;
  padding-left: 0;
  margin: 0;
`

export const LinkWrapper = styled.li`
  display: block;
  font-size: 18px;
  font-weight: 100;
  list-style: none;
  padding: ${({ theme }) => theme.spacing.l} ${({ theme }) => theme.spacing.xl};
  position: relative;
  text-decoration: none;

  & :after {
    content: '';
    position: absolute;
    opacity: 0.5;
    height: 1px;
    width: 85%;
    background-color: ${({ theme }) => theme.colors.white};
    bottom: 0;
    left: 50%;
    transform: translate(-50%);
  }

  & span {
    color: ${({ theme }) => theme.colors.white};
    padding-left: 30px;
  }
  & svg {
    color: ${({ theme }) => theme.colors.white};
  }
`

export const LinkItem = styled.a`
  color: ${({ theme }) => theme.colors.white};
  display: flex;
  align-items: center;
`
