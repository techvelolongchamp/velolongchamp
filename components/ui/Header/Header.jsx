import React from 'react'
import PropTypes from 'prop-types'
import { animateScroll as scroll } from 'react-scroll'
import { useIntl } from 'react-intl'
import Link from 'next/link'

import { useMatches } from '../../../hooks/useMatches'
import LanguageSelect from '../../LanguageSelect'

import theme from '../../../styles/theme'

import {
  HeaderWrapper,
  LeftContainer,
  ScrollTitle,
  TitleLink,
  Logo,
  NavList,
  NavItem,
  ScrollItem,
  RightContainer,
  ButtonLink,
  Wrapper,
  MobileNavList,
  MobileScrollItem,
} from './Header.styled'

const Header = ({ noScroll, handleLink }) => {
  const { formatMessage } = useIntl()
  const isMobile = useMatches('(max-width: 640px)')

  const scrollTop = () => {
    scroll.scrollToTop()
  }

  return (
    <HeaderWrapper id="header">
      <Wrapper>
        <LeftContainer>
          {noScroll ? (
            <Link href="/">
              <TitleLink>
                <Logo src="/logo.png" alt="logo" />
              </TitleLink>
            </Link>
          ) : (
            <ScrollTitle onClick={scrollTop} title="Home">
              <Logo src="/logo.png" alt="logo" />
            </ScrollTitle>
          )}

          {!isMobile && !noScroll && (
            <NavList>
              <li>
                <ScrollItem
                  title={formatMessage({ id: 'header.presentation' })}
                  activeClass="active"
                  smooth
                  spy
                  offset={-theme.navbarHeightNb}
                  to={handleLink[0]}
                >
                  {formatMessage({ id: 'header.presentation' })}
                </ScrollItem>
              </li>
              <li>
                <ScrollItem
                  title={formatMessage({ id: 'header.joinUs' })}
                  activeClass="active"
                  smooth
                  spy
                  offset={-theme.navbarHeightNb}
                  to={handleLink[1]}
                >
                  {formatMessage({ id: 'header.joinUs' })}
                </ScrollItem>
              </li>
              <li>
                <ScrollItem
                  title={formatMessage({ id: 'header.meetUs' })}
                  activeClass="active"
                  smooth
                  spy
                  offset={-theme.navbarHeightNb}
                  to={handleLink[2]}
                >
                  {formatMessage({ id: 'header.meetUs' })}
                </ScrollItem>
              </li>
              <li>
                <ScrollItem
                  title={formatMessage({ id: 'header.contactUs' })}
                  activeClass="active"
                  smooth
                  spy
                  offset={-theme.navbarHeightNb}
                  to={handleLink[3]}
                >
                  {formatMessage({ id: 'header.contactUs' })}
                </ScrollItem>
              </li>
            </NavList>
          )}
        </LeftContainer>
        <RightContainer>
          <nav>
            <NavList>
              <li>
                <Link href="/blog">
                  <NavItem>{formatMessage({ id: 'header.blog' })}</NavItem>
                </Link>
              </li>
            </NavList>
          </nav>
          <Link href="/join">
            <ButtonLink>{formatMessage({ id: 'header.clickJoin' })}</ButtonLink>
          </Link>
          <LanguageSelect />
        </RightContainer>
      </Wrapper>
      {isMobile && !noScroll && (
        <MobileNavList>
          <li>
            <MobileScrollItem
              title={formatMessage({ id: 'header.presentation' })}
              activeClass="active"
              smooth
              spy
              offset={-theme.mobielNavbarHeightNb}
              to={handleLink[0]}
            >
              {formatMessage({ id: 'header.presentation' })}
            </MobileScrollItem>
          </li>
          <li>
            <MobileScrollItem
              title={formatMessage({ id: 'header.joinUs' })}
              activeClass="active"
              smooth
              spy
              offset={-theme.mobielNavbarHeightNb}
              to={handleLink[1]}
            >
              {formatMessage({ id: 'header.joinUs' })}
            </MobileScrollItem>
          </li>
          <li>
            <MobileScrollItem
              title={formatMessage({ id: 'header.meetUs' })}
              activeClass="active"
              smooth
              spy
              offset={-theme.mobielNavbarHeightNb}
              to={handleLink[2]}
            >
              {formatMessage({ id: 'header.meetUs' })}
            </MobileScrollItem>
          </li>
          <li>
            <MobileScrollItem
              title={formatMessage({ id: 'header.contactUs' })}
              activeClass="active"
              smooth
              spy
              offset={-theme.mobielNavbarHeightNb}
              to={handleLink[3]}
            >
              {formatMessage({ id: 'header.contactUs' })}
            </MobileScrollItem>
          </li>
        </MobileNavList>
      )}
    </HeaderWrapper>
  )
}

Header.propTypes = {
  noScroll: PropTypes.bool,
  handleLink: PropTypes.arrayOf(PropTypes.string),
}

export default Header
