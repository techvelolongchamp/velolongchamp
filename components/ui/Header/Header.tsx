import React from 'react'
import { animateScroll as scroll } from 'react-scroll'
import { useIntl } from 'react-intl'
import Link from 'next/link'
import { FaCalendarAlt } from 'react-icons/fa'

import { useMatches } from '../../../hooks/useMatches'
import LanguageSelect from '../../LanguageSelect'

import theme from '../../../styles/theme'

import {
  HeaderWrapper,
  LeftContainer,
  ScrollTitle,
  TitleLink,
  Logo,
  NavList,
  NavItemList,
  NavItem,
  ScrollItem,
  RightContainer,
  ButtonLink,
  Wrapper,
  MobileNavList,
  MobileScrollItem,
} from './Header.styled'

const Header: React.FC<{ noScroll?: boolean; handleLink?: string[] }> = ({
  noScroll,
  handleLink,
}) => {
  const { formatMessage } = useIntl()
  const isMobile = useMatches('(max-width: 640px)')

  const scrollTop = () => {
    scroll.scrollToTop()
  }

  return (
    <HeaderWrapper id="header">
      <Wrapper>
        <LeftContainer>
          {noScroll ? (
            <Link href="/" passHref>
              <TitleLink data-testid="title-link" id="logo-link">
                <Logo src="/logo.png" alt="logo" />
              </TitleLink>
            </Link>
          ) : (
            <ScrollTitle
              id="logo-scroll"
              data-testid="scroll-title"
              onClick={scrollTop}
              title="Home"
            >
              <Logo src="/logo.png" alt="logo" />
            </ScrollTitle>
          )}

          {!isMobile && !noScroll && handleLink && handleLink.length === 4 && (
            <NavList>
              <li data-testid="scroll-item">
                <ScrollItem
                  title={formatMessage({ id: 'header.presentation' })}
                  activeClass="active"
                  smooth
                  spy
                  offset={-theme.navbarHeightNb}
                  to={handleLink[0]}
                  id={handleLink[0] + 'scrolllink'}
                >
                  {formatMessage({ id: 'header.presentation' })}
                </ScrollItem>
              </li>
              <li data-testid="scroll-item">
                <ScrollItem
                  title={formatMessage({ id: 'header.joinUs' })}
                  activeClass="active"
                  smooth
                  spy
                  offset={-theme.navbarHeightNb}
                  to={handleLink[1]}
                  id={handleLink[1] + 'scrolllink'}
                >
                  {formatMessage({ id: 'header.joinUs' })}
                </ScrollItem>
              </li>
              <li data-testid="scroll-item">
                <ScrollItem
                  title={formatMessage({ id: 'header.meetUs' })}
                  activeClass="active"
                  smooth
                  spy
                  offset={-theme.navbarHeightNb}
                  to={handleLink[2]}
                  id={handleLink[2] + 'scrolllink'}
                >
                  {formatMessage({ id: 'header.meetUs' })}
                </ScrollItem>
              </li>
              <li data-testid="scroll-item">
                <ScrollItem
                  title={formatMessage({ id: 'header.contactUs' })}
                  activeClass="active"
                  smooth
                  spy
                  offset={-theme.navbarHeightNb}
                  to={handleLink[3]}
                  id={handleLink[3] + 'scrolllink'}
                >
                  {formatMessage({ id: 'header.contactUs' })}
                </ScrollItem>
              </li>
            </NavList>
          )}
        </LeftContainer>
        <RightContainer>
          <nav>
            <NavItemList>
              <li>
                <Link href="/calendar" passHref>
                  <NavItem id="calendar-link">
                    <FaCalendarAlt />
                  </NavItem>
                </Link>
              </li>
              <li>
                <Link href="/blog" passHref>
                  <NavItem id="blog-link">
                    {formatMessage({ id: 'header.blog' })}
                  </NavItem>
                </Link>
              </li>
            </NavItemList>
          </nav>
          <Link href="/join" passHref>
            <ButtonLink id="join-link">
              {formatMessage({ id: 'header.clickJoin' })}
            </ButtonLink>
          </Link>
          <LanguageSelect />
        </RightContainer>
      </Wrapper>
      {isMobile && !noScroll && handleLink && handleLink.length === 4 && (
        <MobileNavList>
          <li data-testid="mobile-item">
            <MobileScrollItem
              title={formatMessage({ id: 'header.presentation' })}
              activeClass="active"
              smooth
              spy
              offset={-theme.mobielNavbarHeightNb}
              to={handleLink[0]}
              id={handleLink[0] + 'scrolllink'}
            >
              {formatMessage({ id: 'header.presentation' })}
            </MobileScrollItem>
          </li>
          <li data-testid="mobile-item">
            <MobileScrollItem
              title={formatMessage({ id: 'header.joinUs' })}
              activeClass="active"
              smooth
              spy
              offset={-theme.mobielNavbarHeightNb}
              to={handleLink[1]}
              id={handleLink[1] + 'scrolllink'}
            >
              {formatMessage({ id: 'header.joinUs' })}
            </MobileScrollItem>
          </li>
          <li data-testid="mobile-item">
            <MobileScrollItem
              title={formatMessage({ id: 'header.meetUs' })}
              activeClass="active"
              smooth
              spy
              offset={-theme.mobielNavbarHeightNb}
              to={handleLink[2]}
              id={handleLink[2] + 'scrolllink'}
            >
              {formatMessage({ id: 'header.meetUs' })}
            </MobileScrollItem>
          </li>
          <li data-testid="mobile-item">
            <MobileScrollItem
              title={formatMessage({ id: 'header.contactUs' })}
              activeClass="active"
              smooth
              spy
              offset={-theme.mobielNavbarHeightNb}
              to={handleLink[3]}
              id={handleLink[3] + 'scrolllink'}
            >
              {formatMessage({ id: 'header.contactUs' })}
            </MobileScrollItem>
          </li>
        </MobileNavList>
      )}
    </HeaderWrapper>
  )
}

export default Header
