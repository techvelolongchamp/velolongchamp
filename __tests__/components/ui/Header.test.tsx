import React from 'react'

import { RenderWithProvider as render } from '../../provider'

import Header from '../../../components/ui/Header/Header'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const handleLink = ['presentation', 'join', 'meeting', 'contact']

describe('Header unit tests', () => {
  it('should render correctly', () => {
    window.matchMedia = jest.fn().mockImplementation(() => ({
      addListener: jest.fn(),
      removeListener: jest.fn(),
      matches: false,
    }))
    useRouter.mockImplementation(() => ({ locale: 'fr', pathname: '/' }))
    const { container, getByTestId } = render(<Header />)
    const title = getByTestId('scroll-title')
    expect(container).toMatchSnapshot()
    expect(title).toBeInTheDocument()
  })

  it('should render correctly - no scroll', () => {
    window.matchMedia = jest.fn().mockImplementation(() => ({
      addListener: jest.fn(),
      removeListener: jest.fn(),
      matches: false,
    }))
    useRouter.mockImplementation(() => ({ locale: 'fr', pathname: '/' }))
    const { container, getByTestId } = render(<Header noScroll />)

    const title = getByTestId('title-link')
    expect(container).toMatchSnapshot()
    expect(title).toBeInTheDocument()
  })

  it('should render correctly - with links', () => {
    window.matchMedia = jest.fn().mockImplementation(() => ({
      addListener: jest.fn(),
      removeListener: jest.fn(),
      matches: false,
    }))
    useRouter.mockImplementation(() => ({ locale: 'fr', pathname: '/' }))
    const { container, getByTestId, getAllByTestId } = render(
      <Header handleLink={handleLink} />
    )
    const title = getByTestId('scroll-title')
    const scrollItems = getAllByTestId('scroll-item')
    expect(container).toMatchSnapshot()
    expect(title).toBeInTheDocument()
    expect(scrollItems.length).toBe(4)
  })

  it('should render correctly - with links and mobile', () => {
    window.matchMedia = jest.fn().mockImplementation(() => ({
      addListener: jest.fn(),
      removeListener: jest.fn(),
      matches: true,
    }))
    useRouter.mockImplementation(() => ({ locale: 'fr', pathname: '/' }))
    const { container, getByTestId, getAllByTestId } = render(
      <Header handleLink={handleLink} />
    )
    const title = getByTestId('scroll-title')
    const mobileItems = getAllByTestId('mobile-item')
    expect(container).toMatchSnapshot()
    expect(title).toBeInTheDocument()
    expect(mobileItems.length).toBe(4)
  })
})
