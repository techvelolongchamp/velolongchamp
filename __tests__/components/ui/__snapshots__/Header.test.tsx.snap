// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header unit tests should render correctly - no scroll 1`] = `
.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  font-size: 12px;
  margin-left: 5px;
}

.c10 {
  background: none;
  border: none;
  outline: none;
  padding: 5px;
  width: 24px;
}

.c10:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 5;
  background: #f7f7F7;
  padding: 5px 10px;
  border-bottom: 1px solid #fff;
  color: #0d171b;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: 1px solid #6d6e71;
  color: #999999;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 0 20px;
  padding-left: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:hover {
  color: #666666;
}

.c8 {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 10px;
}

.c4 {
  height: 70px;
  padding: 5px 0;
}

@media screen and (max-width:640px) {
  .c7 {
    font-size: 14px;
    border: none;
    padding: 0;
    margin: 0 10px;
  }
}

@media screen and (max-width:640px) {
  .c8 {
    font-size: 14px;
  }
}

<div>
  <header
    class="c0"
    id="header"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          data-testid="title-link"
          href="/"
          id="logo-link"
        >
          <img
            alt="logo"
            class="c4"
            src="/logo.png"
          />
        </a>
      </div>
      <div
        class="c5"
      >
        <nav>
          <ul
            class="c6"
          >
            <li>
              <a
                class="c7"
                href="/blog"
                id="blog-link"
              >
                Actualités
              </a>
            </li>
          </ul>
        </nav>
        <a
          class="c8"
          href="/join"
          id="join-link"
        >
          Adhérer gratuitement
        </a>
        <div
          class="c9"
        >
          <button
            class="c10"
            data-testid="select-en"
            href="/"
            id="select-en"
          >
            EN
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header unit tests should render correctly - with links 1`] = `
.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  font-size: 12px;
  margin-left: 5px;
}

.c11 {
  background: none;
  border: none;
  outline: none;
  padding: 5px;
  width: 24px;
}

.c11:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 5;
  background: #f7f7F7;
  padding: 5px 10px;
  border-bottom: 1px solid #fff;
  color: #0d171b;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: 1px solid #6d6e71;
  color: #999999;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 0 20px;
  padding-left: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:hover {
  color: #666666;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: 1px solid #6d6e71;
  border: none;
  color: #999999;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  height: 30px;
  margin: 0 15px;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:hover {
  color: #666666;
}

.c9 {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 10px;
}

.c4 {
  height: 70px;
  padding: 5px 0;
}

@media screen and (max-width:640px) {
  .c8 {
    font-size: 14px;
    border: none;
    padding: 0;
    margin: 0 10px;
  }
}

@media screen and (max-width:640px) {
  .c9 {
    font-size: 14px;
  }
}

<div>
  <header
    class="c0"
    id="header"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          data-testid="scroll-title"
          id="logo-scroll"
          title="Home"
        >
          <img
            alt="logo"
            class="c4"
            src="/logo.png"
          />
        </a>
        <ul
          class="c5"
        >
          <li
            data-testid="scroll-item"
          >
            <a
              class="c6"
              id="presentationscrolllink"
              title="Présentation"
            >
              Présentation
            </a>
          </li>
          <li
            data-testid="scroll-item"
          >
            <a
              class="c6"
              id="joinscrolllink"
              title="Nous rejoindre"
            >
              Nous rejoindre
            </a>
          </li>
          <li
            data-testid="scroll-item"
          >
            <a
              class="c6"
              id="meetingscrolllink"
              title="Nous rencontrer"
            >
              Nous rencontrer
            </a>
          </li>
          <li
            data-testid="scroll-item"
          >
            <a
              class="c6"
              id="contactscrolllink"
              title="Nous contacter"
            >
              Nous contacter
            </a>
          </li>
        </ul>
      </div>
      <div
        class="c7"
      >
        <nav>
          <ul
            class="c5"
          >
            <li>
              <a
                class="c8"
                href="/blog"
                id="blog-link"
              >
                Actualités
              </a>
            </li>
          </ul>
        </nav>
        <a
          class="c9"
          href="/join"
          id="join-link"
        >
          Adhérer gratuitement
        </a>
        <div
          class="c10"
        >
          <button
            class="c11"
            data-testid="select-en"
            href="/"
            id="select-en"
          >
            EN
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header unit tests should render correctly - with links and mobile 1`] = `
.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  font-size: 12px;
  margin-left: 5px;
}

.c10 {
  background: none;
  border: none;
  outline: none;
  padding: 5px;
  width: 24px;
}

.c10:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 5;
  background: #f7f7F7;
  padding: 5px 10px;
  border-bottom: 1px solid #fff;
  color: #0d171b;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0;
}

.c11 {
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: 1px solid #6d6e71;
  color: #999999;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 0 20px;
  padding-left: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:hover {
  color: #666666;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: 1px solid #6d6e71;
  border: none;
  color: #999999;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  height: 30px;
  margin: 0 15px;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c12:hover {
  color: #666666;
}

.c13 {
  font-size: 14px;
  margin: 0;
  padding: 0 5px;
}

.c8 {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 10px;
}

.c4 {
  height: 70px;
  padding: 5px 0;
}

@media screen and (max-width:640px) {
  .c7 {
    font-size: 14px;
    border: none;
    padding: 0;
    margin: 0 10px;
  }
}

@media screen and (max-width:640px) {
  .c8 {
    font-size: 14px;
  }
}

<div>
  <header
    class="c0"
    id="header"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          data-testid="scroll-title"
          id="logo-scroll"
          title="Home"
        >
          <img
            alt="logo"
            class="c4"
            src="/logo.png"
          />
        </a>
      </div>
      <div
        class="c5"
      >
        <nav>
          <ul
            class="c6"
          >
            <li>
              <a
                class="c7"
                href="/blog"
                id="blog-link"
              >
                Actualités
              </a>
            </li>
          </ul>
        </nav>
        <a
          class="c8"
          href="/join"
          id="join-link"
        >
          Adhérer gratuitement
        </a>
        <div
          class="c9"
        >
          <button
            class="c10"
            data-testid="select-en"
            href="/"
            id="select-en"
          >
            EN
          </button>
        </div>
      </div>
    </div>
    <ul
      class="c6 c11"
    >
      <li
        data-testid="mobile-item"
      >
        <a
          class="c12 c13"
          id="presentationscrolllink"
          title="Présentation"
        >
          Présentation
        </a>
      </li>
      <li
        data-testid="mobile-item"
      >
        <a
          class="c12 c13"
          id="joinscrolllink"
          title="Nous rejoindre"
        >
          Nous rejoindre
        </a>
      </li>
      <li
        data-testid="mobile-item"
      >
        <a
          class="c12 c13"
          id="meetingscrolllink"
          title="Nous rencontrer"
        >
          Nous rencontrer
        </a>
      </li>
      <li
        data-testid="mobile-item"
      >
        <a
          class="c12 c13"
          id="contactscrolllink"
          title="Nous contacter"
        >
          Nous contacter
        </a>
      </li>
    </ul>
  </header>
</div>
`;

exports[`Header unit tests should render correctly 1`] = `
.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  font-size: 12px;
  margin-left: 5px;
}

.c10 {
  background: none;
  border: none;
  outline: none;
  padding: 5px;
  width: 24px;
}

.c10:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 5;
  background: #f7f7F7;
  padding: 5px 10px;
  border-bottom: 1px solid #fff;
  color: #0d171b;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: 1px solid #6d6e71;
  color: #999999;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 0 20px;
  padding-left: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:hover {
  color: #666666;
}

.c8 {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 10px;
}

.c4 {
  height: 70px;
  padding: 5px 0;
}

@media screen and (max-width:640px) {
  .c7 {
    font-size: 14px;
    border: none;
    padding: 0;
    margin: 0 10px;
  }
}

@media screen and (max-width:640px) {
  .c8 {
    font-size: 14px;
  }
}

<div>
  <header
    class="c0"
    id="header"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          data-testid="scroll-title"
          id="logo-scroll"
          title="Home"
        >
          <img
            alt="logo"
            class="c4"
            src="/logo.png"
          />
        </a>
      </div>
      <div
        class="c5"
      >
        <nav>
          <ul
            class="c6"
          >
            <li>
              <a
                class="c7"
                href="/blog"
                id="blog-link"
              >
                Actualités
              </a>
            </li>
          </ul>
        </nav>
        <a
          class="c8"
          href="/join"
          id="join-link"
        >
          Adhérer gratuitement
        </a>
        <div
          class="c9"
        >
          <button
            class="c10"
            data-testid="select-en"
            href="/"
            id="select-en"
          >
            EN
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
